DelvEdit/src/com/interrupt/dungeoneer/editor/EditorApplication.java

```diff
@@ -736,15 +736,30 @@ else if(e == Editor.selection.hovered) {
 
 				end = ray.getEndPoint(rayOutVector, distance + 0.005f);
 
-						// Tile selection bounding
+				// Tile selection bounding
+				boolean isOutOfBounds = false;
 				Editor.selection.tiles.x = (int)end.x;
 				Editor.selection.tiles.y = (int)end.z;
 
-				if(Editor.selection.tiles.x < 0) Editor.selection.tiles.x = 0;
-				if(Editor.selection.tiles.x >= level.width) Editor.selection.tiles.x = level.width - 1;
+				if(Editor.selection.tiles.x < 0) {
```

Comment:
The `TileSelection` object has a `contains()` method and a `getBounds()` method. You might use one of those to simply this logic?

Reply:
Mmh. `contains()` only works on tiles *inside* the selection. Of course I could check if `0` and `widht/height` are part of the selection. `getBounds()` returns the bounds of the selection. So essentially just `x` and `y` coordinates in a struct-like object, so it doesn't change how the algorithm would work. I could create a new method that returns whether the selection is valid/out-of-bounds instead and move the logic in there? But it would be depending on the level properties.

Reply:
Okay. I split it into `outOfLevelBounds()` and `cropToLevelBounds()`. I don't think that additional dependency on the level is too bad. Thoughts?

Reply:
Additionally I overloaded the `clear` method to simplify more logic. I also started adding one-line JavaDoc comments.

Reply:
There are a few points that need clarifying. 
1. TileSelections are collections of tiles. They should make no assumptions about their shape.
2. TileSelections are independent of levels, and can be copy and pasted between levels.

Reply:
Here is a quick hack I implemented in a local copy of your PR:
```java
// Tile selection bounding
Editor.selection.tiles.x = (int)end.x;
Editor.selection.tiles.y = (int)end.z;

TileSelection entireLevel = TileSelection.Rect(0, 0, level.width, level.height);
boolean isOutOfBounds = !entireLevel.contains((int)end.x, (int)end.z);

shouldDrawBox = !isOutOfBounds && shouldDrawBox;
```
In practice the `entireLevel` tile selection only changes when new a level, so it should be a private field on `EditorApplication` or something like that.

Reply:
Ah, okay. That sounds good. I will have a look.